### ISSUE TITLE
Error 260 - tmpstrings not found

### ISSUE THREAD

**Issue by mlostekk** [Flinesoft/BartyCrouch#21]
Hello,

i am trying to use BartyCrouch, but i always get the error 260 while domain is NSCocoaErrorDomain.

I am using this command (all my subfolders like en.lproj with the Localizable.strings files are in the subfolder Localizable. And Beatsnap is the root folder where all code is structures (also subfolders here)

`bartycrouch code -p Beatsnap/ -l Beatsnap/Localizable -a -v`

and getting this response

```
Error Domain=NSCocoaErrorDomain Code=260 "The file “Localizable.strings” couldn’t be opened because there is no such file." UserInfo={NSFilePath=Beatsnap//tmpstrings/Localizable.strings, NSUnderlyingError=0x7f92a2600af0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
Incrementally updated keys of file 'Beatsnap/Localizable/Base.lproj/Localizable.strings'.
Error Domain=NSCocoaErrorDomain Code=260 "The file “Localizable.strings” couldn’t be opened because there is no such file." UserInfo={NSFilePath=Beatsnap//tmpstrings/Localizable.strings, NSUnderlyingError=0x7f92a251ea10 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
Incrementally updated keys of file 'Beatsnap/Localizable/de.lproj/Localizable.strings'.
Error Domain=NSCocoaErrorDomain Code=260 "The file “Localizable.strings” couldn’t be opened because there is no such file." UserInfo={NSFilePath=Beatsnap//tmpstrings/Localizable.strings, NSUnderlyingError=0x7f92a2700ec0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
Incrementally updated keys of file 'Beatsnap/Localizable/en.lproj/Localizable.strings'.
BartyCrouch: Successfully updated strings file(s) of Code files.
```

And even though i added another `NSLocalizedString` and says "Successfully updated..." the file `Beatsnap/Localizable/en.lproj/Localizable.strings` is not changed

**Comment by mlostekk** [comment]
okay, will close it because its related to the "mm" file. Created another issue

**Comment by zbrox** [comment]
I'm still seeing this error and I'm using 3.2.1. Any workarounds or hints on how to solve it?

**Comment by Dschee** [comment]
@zbrox Can you give more information on your setup? The OPs issue was that his files ended on `.mm` and after adding support for `.mm` files himself this was solved for him. Do I understand you correctly that your project has `.mm` files and that the keys there aren't recognized?

Please provide:
- the command you are running (build script content?)
- the hierarchy with all relevant files (containing NSLocalizedString, .strings files)

**Comment by zbrox** [comment]
Sorry for the scarce info. Yes, my project includes .mm files and I'm using the latest release available through cocoapods. When running `bartycrouch code -p "/path/to/code/directory" -l "/directory/containing/all/Localizables" -a`
I get a lot of `Error Domain=NSCocoaErrorDomain Code=260 "The file “Localizable.strings” couldn’t be opened because there is no such file." UserInfo={NSFilePath=[path to code]/tmpstrings/Localizable.strings, NSUnderlyingError=0x7f9aa0f0a4a0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}`

Then when I check the localizable files, they're empty.

**Comment by Dschee** [comment]
I still can't help you without the hierarchy of the relevant files. Specifically it is important not to point directly to the `en.lproj` folders but instead to a folder that contains all of those. Also the path needs to be absolute, e.g. start it with `"$PROJECT_DIR/path/to/localizables"`.

Also note that BartyCrouch isn't available via CocoaPods, but I guess you just mixed it up with Homebrew. ;)

If the hints above don't help you, you can also provide a minimum example project so I can analyze it and fix it. For me it still only sounds like the configuration is incorrect though.

**Comment by zbrox** [comment]
Ah, sorry, I don't know where my head is this Monday morning. I meant Homebrew indeed :)

I'll double check if I'm not using the wrong path. Thanks!

**Comment by zbrox** [comment]
I'm using absolute paths and the pointing to a folder with all the `.lproj` folders which contain the Localizable.strings files. Is this the correct use?

**Comment by Dschee** [comment]
Yeah it should work then. Could you provide a minimum example project? That would really help here. I'm also reopening the issue.

**Comment by zbrox** [comment]
Thanks! I'll try to do this when I have time, hopefully tomorrow.

**Comment by Dschee** [comment]
Ah, one more idea: BartyCrouch writes a temporary file to your Localizables path. Maybe you don't have write access? Usually that shouldn't be the case but I can't come up with something else at the moment.

**Comment by zbrox** [comment]
I think I discovered the problem. I couldn't reproduce it in a sample project until I renamed the project folder to have a space in the folder name.

**Comment by Dschee** [comment]
Oh, if that's the problem, then this must be a duplicate of #12 which was attempted to be fixed via #17. Seems like there's more to it than handled in #17. Let me close this then and open the more descriptive #12 instead.
